git-odb/src/store/loose/write.rs

```diff
@@ -28,6 +29,22 @@ pub enum Error {
 impl crate::write::Write for Store {
     type Error = Error;
 
+    fn write(&self, object: impl WriteTo, hash: git_hash::Kind) -> Result<git_hash::ObjectId, Self::Error> {
+        let mut to = self.dest(hash)?;
+        to.write_all(&object.loose_header()).map_err(|err| Error::Io {
```

Comment:
One more thing: I do feel uneasy with trusting the `size()` implementation just yet, so I'd like to see it run against a moderately sized pack

Fortunately, there is already a program to exercise this code path, it's `gix pack-explode`. This [stress test](https://github.com/Byron/gitoxide/blob/e8b091943f0c9a26317da0003f7fcdf5a56ef21a/Makefile#L263) could be amended by another run that writes actual objects instead of using the sink. It will automatically verify that the written hash matches the one from the pack.

The stress test could be run against the Rustc repository for instance. Maybe there are other ways too though, maybe I am just paranoid. By the time of this writing, 20% of the linux kernel pack objects were exploded to disk without issue.

Reply:
The linux kernel pack could successfully exercise the new code path using computed sizes in the header. It would be nice to see how long exploding the Rust pack to disk takes on CI to evaluate if it's prohibitive or not to keep active.

```
➜  gitoxide git:(xmo-odoo-git-loose-objects) ./target/release/gix --verbose pack-explode tests/fixtures/repos/linux.git/objects/pack/pack-3ee05b0f4e4c2cb59757c95c68e2d13c0a491289.idx out
 06:14:01 SHA1 of index done 212.8MB in 0.13s (1.6GB/s)
 06:14:02  SHA1 of pack done 1.4GB in 0.92s (1.5GB/s)
 06:14:02 collecting sorted index done 7.6M entries in 1.38s (5.5M entries/s)
 06:40:38              Traversing of 7600359 objects done in 1595.78s (4762 objects/s, ~59.9 MB/s)
```

Reply:
> maybe I am just paranoid. By the time of this writing, 20% of the linux kernel pack objects were exploded to disk without issue.

Seems to me you're merely cautious which seems like a good idea.

Testing on linux kernel packs will probably have to wait until next week though: apparently I didn't have a local clone of the linux kernel repository, and the holidays DSL is unable to check it out.

(I managed to find a workaround for the ssh2-rs issue so I could run the test suite, I've an odd failure in `various_termination_signals_remove_tempfiles_unconditionally` but that might be because this machine runs a really old macos because... reasons).